hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -62,16 +63,38 @@ protected void writeNSSummariesToDB(Map<Long, NSSummary> nsSummaryMap)
     try (RDBBatchOperation rdbBatchOperation = new RDBBatchOperation()) {
       for (Map.Entry<Long, NSSummary> entry : nsSummaryMap.entrySet()) {
         try {
-          reconNamespaceSummaryManager.batchStoreNSSummaries(rdbBatchOperation,
-              entry.getKey(), entry.getValue());
+          reconNamespaceSummaryManager.batchStoreNSSummaries(rdbBatchOperation, entry.getKey(), entry.getValue());
         } catch (IOException e) {
-          LOG.error("Unable to write Namespace Summary data in Recon DB.",
-              e);
+          LOG.error("Unable to write Namespace Summary data in Recon DB.", e);
           throw e;
         }
       }
       reconNamespaceSummaryManager.commitBatchOperation(rdbBatchOperation);
     }
+    LOG.debug("Successfully written Namespace Summary data in Recon DB.");
+  }
+
+  protected void deleteNSSummariesFromDB(Collection<Long> objectIdsToBeDeleted) {
+    try {
+      addAndCommitToDeleteNSSummaries(objectIdsToBeDeleted);
```

Comment:
these 2 need refactor with flushAndCommitNSToDB() as comment given below and can be combined.

Reply:
@sumitagrawl Thanks for the review. All review comments addressed. Kindly re-review.